p2p/src/sync/mod.rs

```diff
@@ -162,16 +163,23 @@ where
             Arc::clone(&self.is_initial_block_download),
             self.time_getter.clone(),
         );
-        tokio::spawn(async move {
+        let peer_task = tokio::spawn(async move {
             peer.run().await;
         });
+
+        let prev_task = self.peers.insert(peer_id, peer_task);
+        assert!(prev_task.is_none(), "Registered duplicated peer: {peer_id}");
     }
 
     /// Stops the task of the given peer by closing the corresponding channel.
-    fn unregister_peer(&mut self, peer: PeerId) {
-        log::debug!("Unregister peer {peer} from sync manager");
-        let removed = self.peers.remove(&peer);
-        assert!(removed, "Unregistering unknown peer: {peer}");
+    fn unregister_peer(&mut self, peer_id: PeerId) {
```

Comment:
Are these `register_peer()`/`unregister_peer()` changes supposed to be included in this PR?

Reply:
It's needed to abort async tasks if they are sleeping.